account.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/cockroachdb/apd/v2"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	auth "github.com/cosmos/cosmos-sdk/x/auth/types"
+	vesting "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	bank "github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// Account is an internal representation of a genesis regen account
+type Account struct {
+	Address       sdk.AccAddress
+	TotalRegen    apd.Decimal
+	Distributions []Distribution
+}
+
+// Distribution is an internal representation of a genesis vesting distribution of regen
+type Distribution struct {
+	Time  time.Time
+	Regen apd.Decimal
+}
+
+const (
+	URegenDenom = "uregen"
+)
+
+var tenE6 *apd.Decimal
+
+func init() {
+	var err error
+	tenE6, _, err = apd.NewFromString("1000000")
+	if err != nil {
+		panic(err)
+	}
+}
+
+func ToCosmosAccount(acc Account, genesisTime time.Time) (auth.AccountI, *bank.Balance, error) {
+	totalCoins, err := RegenToCoins(&acc.TotalRegen)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	addrStr := acc.Address.String()
+	balance := &bank.Balance{
+		Address: addrStr,
+		Coins:   totalCoins,
+	}
+
+	if len(acc.Distributions) == 0 {
+		return &auth.BaseAccount{Address: addrStr}, balance, nil
+	}
+
+	startTime := acc.Distributions[0].Time
```

Comment:
Won't this error if `len(acc.Distributions) == 0` ? I'm assuming that can still be the case since you left the `if` block above.

Reply:
No because the previous if checks for len == 0 and returns out